aries_cloudagent/connections/models/diddoc_v2/diddoc.py

```diff
@@ -0,0 +1,266 @@
+    """
+
+    CONTEXT = "https://w3id.org/did/v1"
+
+
+    def __init__(self, did: str = None, id: str = None, alsoKnownAs: list = None,
+                 controller=None, verificationMethod: list = None, authentication=None,
+                 assertionMethod=None, keyAgreement=None, capabilityInvocation=None,
+                 capabilityDelegation=None, publicKey=None, service=None,
+                 json=None) -> None:
+
+        """
+        Initialize the DIDDoc instance.
+
+        Retain DID ('id' in DIDDoc context); initialize verification keys
+        and services to empty lists.
+
+        Args:
+            did: DID for current DIDdoc
+            TODO: complete
+
+        Raises:
+            ValueError: for bad input DID.
+
+        """
+        self._alsoKnownAs = None
+        self._controller = None
+        self._verificationMethod = None
+        self._authentication = None
+        self._assertionMethod = None
+        self._keyAgreement = None
+        self._capabilityInvocation = None
+        self._capabilityDelegation = None
+        self._publicKey = None
+        self._service = None
+
+        if json:
+            did_doc = self.deserialize(json)
+            self.__clone_did_doc__(did_doc, atributes=json.keys())
+            return
+        if did:
+            self._id = canon_did(did) if did else None  # allow specification post-hoc
+
+        elif id:
+            self._id = id
+        else:
+            raise ValueError('did or id are required for DIDDoc instantiation')
+
+        if alsoKnownAs:
+            self._alsoKnownAs = alsoKnownAs
+
+        if controller:
+            self._controller = controller
+
+        if verificationMethod:
+            self._verificationMethod = verificationMethod
+
+        if authentication:
+            self._authentication = authentication
+
+        if assertionMethod:
+            self._assertionMethod = assertionMethod
+
+        if keyAgreement:
+            self._keyAgreement = keyAgreement
+
+        if capabilityInvocation:
+            self._capabilityInvocation = capabilityInvocation
+
+        if capabilityDelegation:
+            self._capabilityDelegation = capabilityDelegation
+
+        if publicKey:
+            self._publicKey = publicKey
+
+        if service:
+            self._service = service
+
+    def deserialize(self, json: dict):
+        """
+        Deserialize a dict into a DIDDoc object.
+
+        Args:
+            json: service or public key to set
+        Returns: DIDDoc object
+        """
+        schema = DIDDocSchema()
+        did_doc = schema.load(json)
+        return did_doc
+
+    def serialize(self) -> dict:
+        """
+        Serialize the DIDDoc object into dict.
+
+        Returns: Dict
+        """
+        schema = DIDDocSchema()
+        did_doc = schema.dump(self)
+        did_doc["@context"] = self.CONTEXT
+        return did_doc
+
+    def __clone_did_doc__(self, did_doc, atributes):
```

Comment:
After removing the `json` `__init__` parameter, I think this method should be removed. The logic here is difficult to follow and I think unneeded when we reorient to the loading as a classmethod.